smithy-model/src/main/java/software/amazon/smithy/model/shapes/EnumShape.java

```diff
@@ -0,0 +1,413 @@
+                    // than a string value. Rather than letting the exception propagate
+                    // here, we let the shape validator handle it because it will give
+                    // a much better error.
+                    return;
+                }
+            }
+            addTrait(builder.build());
+        }
+
+        @Override
+        public ShapeType getShapeType() {
+            return ShapeType.ENUM;
+        }
+
+        @Override
+        public Builder id(ShapeId shapeId) {
+            super.id(shapeId);
+            for (MemberShape member : members.peek().values()) {
+                addMember(member.toBuilder().id(shapeId.withMember(member.getMemberName())).build());
+            }
+            return this;
+        }
+
+        public Builder members(EnumTrait trait, boolean synthesizeNames) {
+            if (getId() == null) {
+                throw new IllegalStateException("An id must be set before adding a named enum trait to a string.");
+            }
+            clearMembers();
+
+            for (EnumDefinition definition : trait.getValues()) {
+                Optional<MemberShape> member = definition.asMember(getId(), synthesizeNames);
+                if (member.isPresent()) {
+                    addMember(member.get());
+                } else {
+                    throw new IllegalStateException(String.format(
+                            "Unable to convert enum trait entry with name: `%s` and value `%s` to an enum member.",
+                            definition.getName().orElse(""), definition.getValue()
+                    ));
+                }
+            }
+
+            return this;
+        }
+
+        public Builder members(EnumTrait trait) {
+            return members(trait, false);
+        }
+
+        /**
+         * Replaces the members of the builder.
+         *
+         * @param members Members to add to the builder.
+         * @return Returns the builder.
+         */
+        public Builder members(Collection<MemberShape> members) {
+            clearMembers();
+            for (MemberShape member : members) {
+                addMember(member);
+            }
+            return this;
+        }
+
+        /**
+         * Removes all members from the shape.
+         *
+         * @return Returns the builder.
+         */
+        public Builder clearMembers() {
+            members.clear();
+            return this;
+        }
+
+        @Override
+        public Builder addMember(MemberShape member) {
+            if (!member.getTarget().equals(UnitTypeTrait.UNIT)) {
+                throw new SourceException(String.format(
+                        "Enum members may only target `smithy.api#Unit`, but found `%s`", member.getTarget()
+                ), getSourceLocation());
+            }
+            if (!member.hasTrait(EnumValueTrait.ID) && !member.hasTrait(EnumDefaultTrait.ID)) {
```

Comment:
It's interesting that this is adding a default EnumValueTrait. That would be good to call out somewhere in javadocs.

A consequence of this is that you couldn't add a selector that warns when `@enumValue` isn't set. Not saying that's necessary, but it's not possible when we add default traits like this.

Reply:
The alternative to making this default is that you always have to check two things. I'm not against it necessarily, especially since we've already got the method to just get a map of name-value pairs.